pkg/signup/service/signup_service.go

```diff
@@ -196,20 +197,37 @@ func (s *ServiceImpl) createUserSignup(ctx *gin.Context) (*v1alpha1.UserSignup,
 }
 
 // reactivateUserSignup reactivates the deactivated UserSignup resource with the specified username and userID
-func (s *ServiceImpl) reactivateUserSignup(ctx *gin.Context, existing v1alpha1.UserSignup) (*v1alpha1.UserSignup, error) {
+func (s *ServiceImpl) reactivateUserSignup(ctx *gin.Context, existing *v1alpha1.UserSignup) (*v1alpha1.UserSignup, error) {
 	// Update the existing usersignup's spec and annotations/labels by new values from a freshly generated one.
 	// We don't want to deal with merging/patching the usersignup resource
 	// and just want to reset the spec and annotations/labels so they are the same as in a freshly created usersignup resource.
 	newUserSignup, err := s.newUserSignup(ctx)
 	if err != nil {
 		return nil, err
 	}
-
+	log.WithValues(map[string]interface{}{"toolchain.dev.openshift.com/activation-counter": existing.Annotations["toolchain.dev.openshift.com/activation-counter"]}).
+		Info(ctx, "reactivating user")
+	// keep the same annotations but preserve (don't override) the `toolchain.dev.openshift.com/activation-counter` one
+	if c, exists := existing.Annotations[v1alpha1.UserSignupActivationCounterAnnotationKey]; exists {
+		if activations, err := strconv.Atoi(c); err == nil {
+			log.WithValues(map[string]interface{}{"value": activations + 1}).
+				Info(ctx, "incrementing the 'toolchain.dev.openshift.com/activation-counter' annotation")
+			newUserSignup.Annotations[v1alpha1.UserSignupActivationCounterAnnotationKey] = strconv.Itoa(activations + 1)
+		} else {
+			log.WithValues(map[string]interface{}{"value": c}).
+				Error(ctx, err, "invalid value for the 'toolchain.dev.openshift.com/activation-counter' annotation")
+			newUserSignup.Annotations[v1alpha1.UserSignupActivationCounterAnnotationKey] = "2" // the user is coming back, but we don't know how many times yet, so let's assume it's the second time
+		}
+	} else {
+		log.WithValues(map[string]interface{}{"value": c}).
+			Info(ctx, "missing value for the 'toolchain.dev.openshift.com/activation-counter annotation")
+		newUserSignup.Annotations[v1alpha1.UserSignupActivationCounterAnnotationKey] = "2" // "best-effort": assume the user signed-up only once before returning now
+	}
```

Comment:
correct me if I'm wrong - this will increment number for all users before passing the phone verification again.
From my point of view, we should increment when we either [approve the user in UserSignup controller](https://github.com/codeready-toolchain/host-operator/blob/master/pkg/controller/usersignup/usersignup_controller.go#L404-L405) or when we provision the MUR again.

Reply:
You are correct Matous. Reg-service just initiates signup "from scratch" here. Verification happens later.
So, we can't increment the metric here. I would move it to the user signup controller too.

Reply:
logic was removed in https://github.com/codeready-toolchain/registration-service/pull/178/commits/08f1ab839664d59d46cc43c7e52746d14bd05316